report_py3o/models/ir_actions_report.py

```diff
@@ -0,0 +1,200 @@
+        )
+        try:
+            lo_bin = find_in_path(lo_bin)
+        except IOError:
+            lo_bin = None
+        return lo_bin
+
+    @api.depends("report_type", "py3o_filetype")
+    def _compute_is_py3o_native_format(self):
+        fmt = Formats()
+        for rec in self:
+            rec.is_py3o_native_format = False
+            if not rec.report_type == "py3o" or not rec.py3o_filetype:
+                continue
+            filetype = rec.py3o_filetype
+            rec.is_py3o_native_format = fmt.get_format(filetype).native
+
+    def _compute_lo_bin_path(self):
+        lo_bin = self._get_lo_bin()
+        for rec in self:
+            rec.lo_bin_path = lo_bin
+
+    @api.depends("lo_bin_path", "is_py3o_native_format", "report_type")
+    def _compute_py3o_report_not_available(self):
+        for rec in self:
+            rec.is_py3o_report_not_available = False
+            rec.msg_py3o_report_not_available = ""
+            if not rec.report_type == "py3o":
+                continue
+            if not rec.is_py3o_native_format and not rec.lo_bin_path:
+                rec.is_py3o_report_not_available = True
+                rec.msg_py3o_report_not_available = (
+                    _(
+                        "The libreoffice runtime is required to genereate the "
+                        "py3o report '%s' but is not found into the bin path. You "
+                        "must install the libreoffice runtime on the server. If "
+                        "the runtime is already installed and is not found by "
+                        "Odoo, you can provide the full path to the runtime by "
+                        "setting the key 'py3o.conversion_command' into the "
+                        "configuration parameters."
+                    )
+                    % rec.name
+                )
+
+    @api.model
+    def get_from_report_name(self, report_name, report_type):
+        return self.search(
+            [("report_name", "=", report_name), ("report_type", "=", report_type)]
+        )
+
+    def _render_py3o(self, res_ids, data):
+        self.ensure_one()
+        if self.report_type != "py3o":
+            raise RuntimeError(
+                "py3o rendition is only available on py3o report.\n"
+                "(current: '{}', expected 'py3o'".format(self.report_type)
+            )
+        return (
+            self.env["py3o.report"]
+            .create({"ir_actions_report_id": self.id})
+            .create_report(res_ids, data)
+        )
+
+    def gen_report_download_filename(self, res_ids, data):
+        """Override this function to change the name of the downloaded report
+        """
+        self.ensure_one()
+        report = self.get_from_report_name(self.report_name, self.report_type)
+        if report.print_report_name and not len(res_ids) > 1:
+            obj = self.env[self.model].browse(res_ids)
+            return safe_eval(report.print_report_name, {"object": obj, "time": time})
```

Comment:
```suggestion
            return safe_eval(report.print_report_name, {"object": obj,})
```
In v 14.0 time module is already in safe_eval https://github.com/odoo/odoo/blob/14.0/odoo/tools/safe_eval.py#L38

Reply:
retreturn ?

Reply:
fixed. thanks